client/azure_client_set.go

```diff
@@ -171,6 +177,20 @@ func newVirtualNetworkClient(config *clientConfig) *network.VirtualNetworksClien
 	return &c
 }
 
+func newVirtualNetworkGatewayConnectionsClient(config *clientConfig) *network.VirtualNetworkGatewayConnectionsClient {
+	c := network.NewVirtualNetworkGatewayConnectionsClientWithBaseURI(config.resourceManagerEndpoint, config.subscriptionID)
```

Comment:
Can't we just use this one: https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-05-01/network#NewVirtualNetworkGatewayConnectionsClient ?

Reply:
This would tight us to only support azure PublicCloud.

The current implementation combinig `azure.EnvironmentFromName` and `New*ClientWithBaseURI` allow us to run on different Azure cloud environment.

Reply:
Ack. Thanks.